tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +95,364 @@ mvn package
+
+## Custom SQL++ Queries
+
+1. Get all airlines by country
 
 ```java
-profileCol.remove(id.toString());;
+
+    @Override
+    public List<Airline> findByCountry(String country, int limit, int offset) {
+      String statement = "SELECT airline.id, airline.type, airline.name, airline.iata, airline.icao, airline.callsign, airline.country FROM `"
+                + dbProperties.getBucketName() + "`.`inventory`.`airline` WHERE country = '" + country + "' LIMIT "
+                + limit + " OFFSET " + offset;
+        return cluster
+                .query(statement, QueryOptions.queryOptions().scanConsistency(QueryScanConsistency.REQUEST_PLUS)
+                        .parameters(JsonObject.create().put("country", country)))
+                .rowsAs(Airline.class);
+
+    }
 ```
 
-> *from Delete method of controllers/ProfileController.java*
+<!-- Explaination -->
 
-## GET Profiles by Searching
+In the above example, we are using the `QueryOptions` class to set the `scanConsistency` to `REQUEST_PLUS` to ensure that the query returns the latest data. We are also using the `JsonObject` class to set the `country` parameter in the query.
 
-[N1QL](https://docs.couchbase.com/dotnet-sdk/current/howtos/n1ql-queries-with-sdk.html) is a powerful query language based on SQL, but designed for structured and flexible JSON documents. We will use a N1QL query to search for profiles with Skip, Limit, and Search options.
+Finally, we are using the `rowsAs` method to return the query results as a list of `Airline` objects.
 
-Navigate to the getProfiles method in the ProfileController class.  This endpoint is different from all of the others because it makes the N1QL query rather than a key-value operation. This means more overhead because the query engine is involved. We did create an [index](https://docs.couchbase.com/server/current/learn/services-and-indexes/indexes/indexing-and-query-perf.html) specific for this query, so it should be performant.
+In the query, we are using the `country` parameter to filter the results by country. We are also using the `limit` and `offset` parameters to limit the number of results returned and to implement pagination.
 
-First, the method signature uses the `@RequestParam` annotation to get the individual `skip`, `limit`, and `search` values.  Then, we build our N1QL query using the parameters that were passed in.
+Once the query is executed, the `AirlineController` constructs an HTTP response with a status code of 200 OK and includes the list of airlines in the response body as a list of JSON objects.
 
-Finally, we pass that `query` to the `cluster.query` method and return the result.
+<!-- Link to docs -->
 
-Take notice of the N1QL syntax and how it targets the `bucket`.`scope`.`collection`.
+For more information on the `QueryOptions` class, see the [Couchbase Java SDK documentation](https://docs.couchbase.com/java-sdk/current/howtos/n1ql-queries-with-sdk.html).
```

Comment:
Link the methods above instead of an additional paragraph.

Reply:
removed from here